### PR TITLE
Feature/Date Picker Components

### PR DESCRIPTION
**Pull request by Kelvin-GohKengBoon**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

# Screenshots (if applicable)

Please add appropriate screenshots.

### CODE REVIEW

File: src/DatePicker/index.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import {
+  default as AntDatePicker,
+  DatePickerProps as AntDatePickerProps,
+  RangePickerProps as AntRangePickerProps
+} from 'antd/lib/date-picker'
+
+import '../assets/style/form-item.css'
+import 'antd/lib/style/default.css'
+import 'antd/lib/date-picker/style/index.css'
+import 'antd/lib/empty/style/index.css'
```
Comment (AladdinNullptr) [inline src/DatePicker/index.tsx:11]: There is a dependency on the empty component. Will need to include the new Empty component styles here

File: src/DatePicker/stories/index.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
+import React from 'react'
+import { ComponentStory, ComponentMeta } from '@storybook/react'
+import { Form, FormItemProps } from 'antd'
+import moment from 'moment'
```
Comment (AladdinNullptr) [inline src/DatePicker/stories/index.stories.tsx:4]: moment is not maintained anymore. Can use date-fns

File: src/DatePicker/stories/rangePicker.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
+import React from 'react'
+import { ComponentStory, ComponentMeta } from '@storybook/react'
+import { Form, FormItemProps } from 'antd'
+import moment from 'moment'
```
Comment (AladdinNullptr) [inline src/DatePicker/stories/rangePicker.stories.tsx:4]: Same as above

File: src/DatePicker/style.css [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+    @apply px-4 text-slate-300;
+  }
+
+  &-suffix {
+    @apply text-slate-700;
+  }
+
+  &-clear > span > svg{
+    @apply h-4 w-4 text-transparent;
+    background-image: url("data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' width='16' height='16' fill='dimgray' viewBox='0 0 256 256'><rect width='256' height='256' fill='none'></rect><line x1='200' y1='56' x2='56' y2='200' stroke='dimgray' stroke-linecap='round' stroke-linejoin='round' stroke-width='16'></line><line x1='200' y1='200' x2='56' y2='56' stroke='dimgray' stroke-linecap='round' stroke-linejoin='round' stroke-width='16'></line></svg>");
```
Comment (AladdinNullptr) [inline src/DatePicker/style.css:33]: The url can be moved to a `variables.css` file located inside the `src` folder.  Inside it you can define:
`--ic-clear-16: url(...);`

Then in your css:
```
@import '../variables.css';
background-image: var(--ic-clear-16);
```

File: src/DatePicker/style.css [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+    @apply bg-slate-700;
+  }
+
+  &-cell-in-view.ant-picker-cell-in-range {
+    @apply text-white;
+  }
+
+  &-header-view button {
+    @apply text-[#1D3557] text-base font-inter font-bold uppercase leading-6;
+    @apply hover:opacity-80 hover:text-[#1D3557];
```
Comment (AladdinNullptr) [inline src/DatePicker/style.css:72]: You can use preset colors like `text-slate-700`. If the color is not included, it needs to be added in the preset repository

File: src/DatePicker/style.css [hunk 6]
```diff
@@ -0,0 +1,216 @@
...
+    &-suffix {
+      @apply text-red-600;
+    }
+  }
+}
+
+/* sunday */
+.ant-picker-body > table > thead > tr th:nth-child(1) {
+  text-indent: -9999px;
+  line-height: 0; /* Collapse the original line */
```
Comment (AladdinNullptr) [inline src/DatePicker/style.css:151]: Please use tailwindcss classes

File: src/DatePicker/style.css [hunk 7]
```diff
@@ -0,0 +1,216 @@
...
+}
+.ant-picker-body > table > thead > tr th:nth-child(2):after {
+  content: 'M';
+  @apply block font-normal indent-0 leading-normal;
+}
+
+/* tuesday */
+.ant-picker-body > table > thead > tr th:nth-child(3) {
+  text-indent: -9999px;
+  line-height: 0; /* Collapse the original line */
```
Comment (AladdinNullptr) [inline src/DatePicker/style.css:171]: Same as above

File: src/DatePicker/style.css [hunk 8]
```diff
@@ -0,0 +1,216 @@
...
+}
+.ant-picker-body > table > thead > tr th:nth-child(3):after {
+  content: 'T';
+  @apply block font-normal indent-0 leading-normal;
+}
+
+/* wednesday */
+.ant-picker-body > table > thead > tr th:nth-child(4) {
+  text-indent: -9999px;
+  line-height: 0; /* Collapse the original line */
```
Comment (AladdinNullptr) [inline src/DatePicker/style.css:181]: Same as above

File: src/DatePicker/style.css [hunk 9]
```diff
@@ -0,0 +1,216 @@
...
+}
+.ant-picker-body > table > thead > tr th:nth-child(4):after {
+  content: 'W';
+  @apply block font-normal indent-0 leading-normal;
+}
+
+/* thursday */
+.ant-picker-body > table > thead > tr th:nth-child(5) {
+  text-indent: -9999px;
+  line-height: 0; /* Collapse the original line */
```
Comment (AladdinNullptr) [inline src/DatePicker/style.css:191]: Same as above

File: src/DatePicker/style.css [hunk 10]
```diff
@@ -0,0 +1,216 @@
...
+}
+.ant-picker-body > table > thead > tr th:nth-child(5):after {
+  content: 'T';
+  @apply block font-normal indent-0 leading-normal;
+}
+
+/* friday */
+.ant-picker-body > table > thead > tr th:nth-child(6) {
+  text-indent: -9999px;
+  line-height: 0; /* Collapse the original line */
```
Comment (AladdinNullptr) [inline src/DatePicker/style.css:201]: Same as above

File: src/DatePicker/style.css [hunk 11]
```diff
@@ -0,0 +1,216 @@
...
+}
+.ant-picker-body > table > thead > tr th:nth-child(6):after {
+  content: 'F';
+  @apply block font-normal indent-0 leading-normal;
+}
+
+/* saturday */
+.ant-picker-body > table > thead > tr th:nth-child(7) {
+  text-indent: -9999px;
+  line-height: 0; /* Collapse the original line */
```
Comment (AladdinNullptr) [inline src/DatePicker/style.css:211]: Same as above

File: src/DatePicker/style.css (hunk 7 above)
Comment (AladdinNullptr) [reply src/DatePicker/style.css:171]: Maybe you can chain chain `nth-child` to reduce repetitive code:
`nth-child(n+4):nth-child(-n+8)`

File: src/DatePicker/stories/index.stories.tsx (hunk 2 above)
Comment (Kelvin-GohKengBoon) [reply src/DatePicker/stories/index.stories.tsx:4]: default type of the Date Picker props : disabledDate is Moment.

File: src/DatePicker/style.css [hunk 12]
```diff
@@ -0,0 +1,200 @@
...
+    @apply text-white;
+  }
+
+  &-header-view button {
+    @apply text-slate-700 text-base font-inter font-bold uppercase leading-6;
+    @apply hover:opacity-80 hover:text-slate-700;
+  }
+
+  &-header-view button:not(:first-child) {
+    margin-left: 4px;
```
Comment (alex-safian) [inline src/DatePicker/style.css:82]:
```suggestion
    @apply ml-1;
```

File: src/DatePicker/style.css [hunk 13]
```diff
@@ -0,0 +1,200 @@
...
+  }
+
+  &-cell-in-view.ant-picker-cell-range-start:not(.ant-picker-cell-range-start-single):not(.ant-picker-cell-range-end)
+    .ant-picker-cell-inner {
+    @apply rounded-tl-2xl rounded-bl-2xl;
+  }
+
+  &-cell-in-view.ant-picker-cell-range-end:not(.ant-picker-cell-range-end-single):not(.ant-picker-cell-range-start)
+    .ant-picker-cell-inner {
+    @apply rounded-tr-2xl rounded-br-2xl;
```
Comment (alex-safian) [inline src/DatePicker/style.css:122]:
```suggestion
    @apply rounded-r-2xl;
```

File: src/DatePicker/style.css [hunk 14]
```diff
@@ -0,0 +1,200 @@
...
+  }
+}
+
+.ant-form-item-label {
+  @apply pb-2 mb-0;
+}
+
+.ant-form-item .ant-form-item-label {
+  @apply leading-none;
+}
```
Comment (alex-safian) [inline src/DatePicker/style.css:132]: We shouldn't have these rules here. If it is a general rule, you must add it in form item styles file. If it is this component specifically, you can add it here by referencing the component class.

File: src/DatePicker/style.css [hunk 15]
```diff
@@ -0,0 +1,200 @@
...
+.ant-picker-body > table > thead > tr th:nth-child(n + 1):nth-child(-n + 8) {
+  @apply indent-[9999px] leading-[0];
+}
+
+.ant-picker-body > table > thead > tr th:nth-child(n + 1):nth-child(-n + 8):after {
+  @apply block font-normal indent-0 leading-normal;
+}
+
+/* sunday */
+.ant-picker-body > table > thead > tr th:nth-child(1):after {
```
Comment (alex-safian) [inline src/DatePicker/style.css:167]: You don't need to have a long selector here. Please check if you can use a short selector rules and apply it to other places too.

File: src/DatePicker/index.tsx [hunk 16]
```diff
@@ -0,0 +1,22 @@
...
+en.lang.quarterFormat = "'Q'Q"
+
+const DatePicker = generatePicker<Date>(dateFnsGenerateConfig)
+
+export const QuarterPicker = DatePicker.QuarterPicker
+
+DatePicker.defaultProps = {
+  suffixIcon: <CaretDown color="#737373" />
+}
+
```
Comment (AladdinNullptr) [inline src/DatePicker/index.tsx:17]: Can add defaultProps for quarter picker as well. There you can add:
`format: "YYYY-'Q'Q"`

File: src/index.tsx [hunk 17]
```diff
@@ -4,4 +4,5 @@ export { default as AvatarGroup, AvatarGroupProps } from './AvatarGroup';
 export { default as Button } from './Button';
 export { default as Radio } from './Radio';
 export { default as Tag } from './Tag';
+export { default as DatePicker } from './DatePicker'
```
Comment (AladdinNullptr) [inline src/index.tsx:7]: need to re-export everything you exported in `index.tsx`

File: src/DatePicker/index.tsx [hunk 18]
```diff
@@ -0,0 +1,32 @@
...
+const DatePicker = generatePicker<Date>(dateFnsGenerateConfig)
+
+export const QuarterPicker = DatePicker.QuarterPicker
+export const WeekPicker = DatePicker.WeekPicker
+export const MonthPicker = DatePicker.MonthPicker
+export const YearPicker = DatePicker.YearPicker
+export const RangePicker = DatePicker.RangePicker
+export const TimePicker = DatePicker.TimePicker
+
+DatePicker.QuarterPicker.defaultProps = {
```
Comment (AladdinNullptr) [inline src/DatePicker/index.tsx:20]:
```suggestion
QuarterPicker.defaultProps = {
```

File: src/DatePicker/stories/index.stories.tsx [hunk 19]
```diff
@@ -0,0 +1,69 @@
...
+    label: 'Date'
+  }
+}
+
+export const BasicQuarter = Template.bind({})
+
+BasicQuarter.args = {
+  datePicker: {
+    label: 'DatePicker',
+    picker: 'quarter'
```
Comment (AladdinNullptr) [inline src/DatePicker/stories/index.stories.tsx:49]: Create a different template for the quarter picker and use the `QuarterPicker` exported from index.ts

File: src/DatePicker/style.css [hunk 20]
```diff
@@ -0,0 +1,194 @@
+@import '../assets/style/form-item.css';
```
Comment (AladdinNullptr) [inline src/DatePicker/style.css:1]: This belongs to the story

**Review by AladdinNullptr — APPROVED** [review]

**Pull request merged**